### ISSUE TITLE
Consider adding a simple Redis SessionRepository implementation

### ISSUE THREAD

**Issue by vpavic** [spring-projects/spring-session#1278]
A lot of complexity in `RedisOperationsSessionRepository` is focused around support for Servlet API's `HttpSessionEvent`. Since many of users do not actually use session events, we can consider providing an alternative, simpler implementation of Redis based `SessionRepository` that wouldn't support session event handling.

/cc @rwinch

**Comment by herveDarritchon** [comment]
👍 

I am very very interested in a simpler repo which doesn't use any events/messaging (pub/sub REDIS stuff) because we have Dynomite over REDIS to insure the cross datacente replication and Dynomite doesn't support pub/sub :)
I tried to do it myself but I am stuck in an issue around redefining the sessionRepository bean with my custom implementation.
I ask on gitter and stackoverflow (https://stackoverflow.com/questions/55818188/custom-spring-session-repository-redis-with-dynomite) for help with no solution.
If anyone can help me, it would be great.

I have difficulties to override the sessionRepository implementation from org.springframework.session:spring-session-data-redis with my custom one.

I tried to set up a config class such as : 

```
@Configuration
@EnableRedisHttpSession
public class SessionConfig extends AbstractHttpSessionApplicationInitializer {

    @Bean
    @Primary
    public SessionRepository sessionRepository() {
        return new RedisDynoSessionRepository();
    }
}
```

And the signature of my Custom Implementation of the SessionRepository is :

```
public class RedisDynoSessionRepository implements
        FindByIndexNameSessionRepository<RedisDynoSessionRepository.RedisSession> {
```

I tried to put the override bean configuration (`spring.main.allow-bean-definition-overriding=true`) or not with no success.
